Assets/UnitInteraction.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+
+public class UnitInteraction : MonoBehaviour {
+
+    private void OnTriggerEnter(Collider other) {
```

Comment:
explain this

Reply:
`OnTrigger` is called when the unit enters a stronghold. The `Collider` in this case is the unit. I added a tag to all Units, and I check that to make sure the interaction is in fact a Unit interaction.

I get the source from the unit _(we set the source of the unit in the game engine above)_ and make sure this is not a collision caused by the unit colliding with the stronghold it came from which happens while the Unit is leaving the stronghold.

Then it calls `ComputeResult()` which updates the owner of this stronghold and the it's new magnitude.